### PR TITLE
support CONV() function

### PR DESCRIPTION
**Pull request by jennifersp**
Fix for https://github.com/dolthub/dolt/issues/4099

### CODE REVIEW

File: sql/expression/function/conv_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		{"toBase is nil", sql.LongText, sql.NewRow('a', 16, nil), nil, nil},
+		{"invalid n", sql.LongText, sql.NewRow("r", 16, 2), "0", nil},
+		{"invalid fromBase", sql.LongText, sql.NewRow(2, 37, 2), nil, nil},
+		{"invalid toBase", sql.LongText, sql.NewRow(2, 16, 37), nil, nil},
+		{"base 16 to base 2", sql.LongText, sql.NewRow("a", 16, 2), "1010", nil},
+		{"base 2 to base 16", sql.LongText, sql.NewRow(1010, 2, 16), "A", nil},
+		{"base 18 to base 8", sql.LongText, sql.NewRow("6E", 18, 8), "172", nil},
+		{"base 8 to base 18", sql.LongText, sql.NewRow("172", 8, 18), "6E", nil},
+		{"base 10 to base -18", sql.LongText, sql.NewRow("-17", 10, -18), "-H", nil},
+		{"base -18 to base 10", sql.LongText, sql.NewRow("-H", -18, 10), "-17", nil},
```
Comment (fulghum) [inline sql/expression/function/conv_test.go:46]: The handling of negative values with conv is a little tricky and didn't match up with how I initially read the MySQL docs.

For example, here's the MySQL results for this test:
```
mysql> select conv("-H", -18, 10);
+----------------------+
| conv("-H", -18, 10)  |
+----------------------+
| 18446744073709551599 |
+----------------------+
```

It would be good to add some more tests around the negative toBase/fromBase value handling. I did some quick testing and it seems like by default a negative int N is cast to an unsigned 64-bit value, which matches the docs. However... I *think* `from_base` might always use the absolute value. I didn't see behavior difference when I used negative/positive values there. `to_base` seems like is what controls if `N` is interpreted as a signed or unsigned value (despite what the docs say about `from_base` controlling this). 

Add some more tests around this handling to make sure we're matching MySQL's behavior here.

File: sql/expression/function/conv_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		{"base -18 to base 10", sql.LongText, sql.NewRow("5276", 8, 16), "ABE", nil},
+		{"base 10 to base 3", sql.LongText, sql.NewRow(123456, 10, 3), "20021100110", nil},
+		{"base 3 to base 10", sql.LongText, sql.NewRow(20021100110, 3, 10), "123456", nil},
+		{"n as hex", sql.LongText, sql.NewRow(0x0a, 10, 10), "10", nil},
```
Comment (fulghum) [inline sql/expression/function/conv_test.go:50]: Another good edge case to test would be an invalid `N` value. Maybe `conv("11B", 10, 10)` and `conv("B11", 10, 10)`.

File: sql/expression/function/conv.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Conv function converts numbers between different number bases. Returns a string representation of the number N, converted from base from_base to base to_base.
```
Comment (fulghum) [inline sql/expression/function/conv.go:26]: Nice comment! 😁 This will be helpful for new eyes to our codebase in the future 😉

**Review by fulghum — COMMENTED** [review]
Looks very close. I found a couple of tricky test edge cases to cover, then I think this is ready.

**Review by fulghum — APPROVED** [review]
Great job with those test cases! Those look really thorough now. 🎉
